### ISSUE TITLE
Mouse lagging with adaptive sync enabled on two monitors

### ISSUE THREAD

**Issue by ferdinandschober** [swaywm/sway#7411]
### Please fill out the following:
- **Sway Version:**
  - `sway version 1.8`

- **Debug Log:**
  - Run `sway -d 2> ~/sway.log` from a TTY and upload it to a pastebin, such as gist.github.com.
  - This will record information about sway's activity. Please try to keep the reproduction as brief as possible and exit sway.
  - Attach the **full** file, do not truncate it.

- **Configuration File:**
  - Default config + 
  ```
  output DP-1 res 2560x1440@240Hz pos 0 0 scale 1 adaptive_sync on
  output DP-2 res 2560x1440@144Hz pos 2560 0 scale 1 adaptive_sync on
  ```

- **Description:**
    - I have two monitors: One with 144hz and one with 240hz refreshrate
    - Both monitors have adaptive sync capabilities
    - When moving the mouse around on the 240hz monitor, it is lagging - feels like ~48 fps
    - Moving the mouse on the lower refreshrate monitor is fine.
    - Disabling adaptive sync on the 240hz monitor fixes the issue (DP-1)

**Comment by ferdinandschober** [comment]
Tell me, if a debug log could be useful

**Comment by YellowOnion** [comment]
Do you have a read out of the FPS from the monitor when this is happening? My monitor will idle at 48fps but ramp up to 180 when I move the mouse (it's also set to 500hz polling), the monitor has a built in OSD FPS meter driven by adaptive-sync rate, it'll help us understand if the monitor is running at 48fps or if it's something else like a pacing issue.

**Comment by ferdinandschober** [comment]
I just retested it and the issue seems to be gone. It reads 125hz on the ODS with the 125hz polling rate mouse, which makes perfect sense!
Guess you can close the issue.

**Comment by feschber** [comment]
https://github.com/swaywm/sway/assets/40996949/bde7848a-8c05-4883-a2bb-86210f2e5f1c


https://github.com/swaywm/sway/assets/40996949/fa0def13-fdc5-4e0e-b3a7-9119747977db

I just noticed this issue seems to persist the two attached files are

a) movement when adaptive sync is enabled, first with 125hz polling rate than at 1000Hz polling rate. The 125Hz polling rate causes lag ONLY on the cursor (as can be seen very clearly)

b) movement with adaptive sync disabled, again 125hz first and 1000hz second - notice that both are now smooth

**Comment by feschber** [comment]
```
swaymsg -t get_version
sway version 1.8.1
```

**Comment by xeome** [comment]
I'm having the same issue but with a single screen, cursor only lags visually but stuff like dragging windows work fine, only happens when I go above 60hz with freesync

**Comment by YellowOnion** [comment]
There's a [known bug in AMD drivers](https://gitlab.freedesktop.org/drm/amd/-/issues/2186) that drop cursor updates when VRR is enabled, that's probably it...and the fact it doesn't show with a low polling rate is probably why I've never personally experienced it.

**Comment by feschber** [comment]
> There's a [known bug in AMD drivers](https://gitlab.freedesktop.org/drm/amd/-/issues/2186) that drop cursor updates when VRR is enabled, that's probably it...and the fact it doesn't show with a low polling rate is probably why I've never personally experienced it.

That could very well be the case. Does that affect KDE as well? I need to test.

**Comment by myghi63** [comment]
> > There's a [known bug in AMD drivers](https://gitlab.freedesktop.org/drm/amd/-/issues/2186) that drop cursor updates when VRR is enabled, that's probably it...and the fact it doesn't show with a low polling rate is probably why I've never personally experienced it.
> 
> That could very well be the case. Does that affect KDE as well? I need to test.

Yes, it does. See: https://gitlab.freedesktop.org/drm/amd/-/issues/2186#note_2289564
